src/main/java/net/fabricmc/filament/task/JavadocLintTask.java

```diff
@@ -0,0 +1,169 @@
+package net.fabricmc.filament.task;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+import cuchaz.enigma.ProgressListener;
+import cuchaz.enigma.translation.mapping.EntryMapping;
+import cuchaz.enigma.translation.mapping.serde.MappingFileNameFormat;
+import cuchaz.enigma.translation.mapping.serde.MappingParseException;
+import cuchaz.enigma.translation.mapping.serde.MappingSaveParameters;
+import cuchaz.enigma.translation.mapping.serde.enigma.EnigmaMappingsReader;
+import cuchaz.enigma.translation.mapping.tree.EntryTree;
+import cuchaz.enigma.translation.representation.entry.Entry;
+import cuchaz.enigma.translation.representation.entry.LocalVariableEntry;
+import cuchaz.enigma.translation.representation.entry.MethodEntry;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.FileType;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.work.ChangeType;
+import org.gradle.work.FileChange;
+import org.gradle.work.Incremental;
+import org.gradle.work.InputChanges;
+import org.gradle.workers.WorkAction;
+import org.gradle.workers.WorkParameters;
+import org.gradle.workers.WorkQueue;
+import org.gradle.workers.WorkerExecutor;
+
+import net.fabricmc.filament.util.FileUtil;
+
+public abstract class JavadocLintTask extends DefaultTask {
+	private static final Pattern PARAM_DOC_LINE = Pattern.compile("^@param\\s+[^<].*$");
+	private final DirectoryProperty mappingDirectory = getProject().getObjects().directoryProperty();
+
+	@Incremental
+	@InputDirectory
+	public DirectoryProperty getMappingDirectory() {
```

Comment:
I think you can just make this abstract. and remove the related field.

Reply:
TIL. Does it need an `@Inject` like `WorkerExecutor`?

Reply:
I dont think so. Just one of the input or output annotations.